src/AutoMapper/Internal/ExpressionFactory.cs

```diff
@@ -56,18 +56,37 @@ public static Expression ForEach(Expression collection, ParameterExpression loop
 
             var breakLabel = Label("LoopBreak");
 
+            Expression disposeCall;
+            if (enumeratorType.GetInterface("System.IDisposable", true) == null)
+            {
+                // Try to dispose at runtime
+                var disposableVar = Variable(typeof(IDisposable), "disposable");
+                disposeCall = Block(new[] { disposableVar },
+                    Assign(disposableVar, TypeAs(enumeratorVar, typeof(IDisposable))),
+                    IfThen(NotEqual(disposableVar, Constant(null, typeof(object))),
+                        Call(disposableVar, disposableVar.Type.GetInheritedMethod(nameof(IDisposable.Dispose)))));
+
+            }
+            else
+            {
+                // Always dispose directly
+                disposeCall = Call(enumeratorVar, enumeratorVar.Type.GetInheritedMethod(nameof(IDisposable.Dispose)));
+            }
```

Comment:
There is no need for strings here.

Reply:
Ok, so how do I call the `Dispose` method?